### PR TITLE
Added dialog validation docs

### LINKED ISSUE #2230: Add docs for CKEDITOR.dialog.validate
## Type of report

Task

## Provide description of the task

`CKEDITOR.dialog.validate` namespace contains many useful methods for validating dialogs, yet there are undocumented. It'd be nice to add even some basic API docs.

## Other details

* Installed CKEditor plugins: dialog

### PR DESCRIPTION
**Pull request by jacekbogdanski**
## What is the purpose of this pull request?

Task

## What changes did you make?

Added docs for `CKEDITOR.dialog.validation`.

Closes #2230

### CODE REVIEW

File: plugins/dialog/plugin.js [hunk 1]
```diff
@@ -3034,16 +3034,33 @@ CKEDITOR.DIALOG_STATE_BUSY = 2;
...
+		 * @class
+		 * @singleton
+		 */
 		CKEDITOR.dialog.validate = {
+			/**
+			 * Performs validation functions composition.
+			 *
+			 * ```
+			 * CKEDITOR.dialog.validation.functions(
+			 * 	CKEDITOR.dialog.validation.notEmpty( 'Value is required.' ),
```
Comment (Comandeer) [inline plugins/dialog/plugin.js]: Incorrect doc block type (we use indentation-based ones) and code indentation (should be indented using <kbd>Tab</kbd>).

File: plugins/dialog/plugin.js [hunk 2]
```diff
@@ -3034,16 +3034,33 @@ CKEDITOR.DIALOG_STATE_BUSY = 2;
...
+			 * 	CKEDITOR.dialog.validation.number( 'Value is not a number.' )
+			 * );
+			 * ```
+			 *
+			 * @param {Mixed[]} arguments
```
Comment (Comandeer) [inline plugins/dialog/plugin.js]: Parameters should be described.

File: plugins/dialog/plugin.js [hunk 3]
```diff
@@ -3034,16 +3034,33 @@ CKEDITOR.DIALOG_STATE_BUSY = 2;
...
+			 * @returns {Function}
```
Comment (Comandeer) [inline plugins/dialog/plugin.js]: Return value also should be described.

File: plugins/dialog/plugin.js [hunk 4]
```diff
@@ -3031,19 +3031,71 @@ CKEDITOR.DIALOG_STATE_BUSY = 2;
...
+		 *
+		 * @class
+		 * @singleton
+		 */
 		CKEDITOR.dialog.validate = {
+			/**
+			 * Performs validation functions composition.
+			 *
+			 * ```javascript
+			 * CKEDITOR.dialog.validation.functions(
```
Comment (Comandeer) [inline plugins/dialog/plugin.js]: It's `CKEDITOR.dialog.validate`, not ~~`CKEDITOR.dialog.validation`~~.

File: plugins/dialog/plugin.js [hunk 5]
```diff
@@ -3031,19 +3031,71 @@ CKEDITOR.DIALOG_STATE_BUSY = 2;
...
+			 * ```
+			 *
+			 * @param {Mixed...} arguments
+			 * Accepts multiple functions which will be composed into a single validation function.
+			 *
+			 * ```javascript
+			 * CKEDITOR.dialog.validation.functions( validator1, validator2 );
+			 * ```
+			 *
+			 * Also you can pass two optional parameters:
```
Comment (Comandeer) [inline plugins/dialog/plugin.js]: If these values are really accepted at the very end of the parameters list, then probably it'll be better to list three parameters, analogically to https://ckeditor.com/docs/ckeditor4/latest/api/CKEDITOR_tools.html#method-extend

Comment (Comandeer) [reply plugins/dialog/plugin.js]: In such case `Mixed...` will become `Function...`

**Comment by jacekbogdanski** [comment]
Somehow Umberto is not able to resolve `@param {CKEDITOR.VALIDATE_OR/CKEDITOR.VALIDATE_AND` so I added info about these enums in a description.

**Pull request merged**